ERCS/erc-7812.md

```diff
@@ -0,0 +1,514 @@
+---
+eip: 7812
+title: ZK Identity Registry
+description: Singleton registry system for storing abstract provable identity statements.
+author: Artem Chystiakov (@arvolear) <artem@rarilabs.com>, Oleksandr Kurbatov <oleksandr@rarilabs.com>, Yaroslav Panasenko <yaroslav@rarilabs.com>, Michael Elliot (@michaelelliot) <mike@zkpassport.id>, Vitalik Buterin (@vbuterin)
+discussions-to: https://ethereum-magicians.org/t/erc-7812-zk-identity-registry/21624
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-11-08
+---
+
+## Abstract
+
+This EIP introduces an on-chain registry system for storing abstract statements, where the state of the system can be proven in zero knowledge without disclosing anything about these statements. Developers may use the singleton `EvidenceRegistry` contract available at `0x<TODO>` to integrate custom business-specific registrars for statement processing and proving.
+
+## Motivation
+
+The standardization and aggregation of provable statements in a singleton on-chain registry significantly improves reusability, scalability, and security of the abundance of zero knowledge privacy-oriented solutions. The abstract specification of the registry allows custom indentity-based, reputation-based, proof-of-attendance-based, etc., protocols to be implemented with little to minimal constraints.
```

Comment:
What is the benefit of centralising these generic provable statements into a single registry? How does it ease the implementation of the other things?

Versus for example, a separate register using similar code via a library? It seems like there would need to be some interaction between the different statement/registry types for this to be of benefit.

More concretely, as a user, if I was wishing to prove a claim based on a passport, what benefit is there to me that the same registry also stores POAP claims?

Reply:
Hey @silasdavis, may you please repost this on the [forum](https://ethereum-magicians.org/t/erc-7812-zk-identity-registry/21624/3) so we could discuss it in more details?

Reply:
Added the rationalization and benefits of singleton design in the rationale section.